contents/javascript/VueJs/Intro.md

```diff
@@ -0,0 +1,85 @@
+# VueJs
+
+Author: Lu Lechuan
+
+A JavaScript framework is an application framework written in JavaScript. It differs from a JavaScript library in its control flow: a library offers functions to be called by its parent code, whereas a framework defines the entire application design.
+JavaScript frameworks aim to improve code quality and maintainablity, example of some javascript frameworks include Angular, React and Vue.
+
+## Advantages of VueJs
+VueJs is an approachable, versetile and performant framework that helps to create a more maintainable and testable database.
+
+* approachable: VueJs is very easy to learn. Comparing to other framework such as Augular, Vue is simple in term of API and design. Learning enough to build non-trivial applications typically takes less than a day.
+
+* progressive: Vue is designed from the ground up to be incrementally adoptable. The core library is focused on the view layer only, and is easy to pick up and integrate with other libraries or existing projects. This means that if you hava a large application, you can plug Vue into just a part of your application without messing the rest up.
+
+* versetile: Vue is perfectly capable of powering sophisticated Single-Page Applications when used in combination with modern tooling and supporting libraries.
+
+* clean: Vue symtax is simple and this can make the HTML pages very clean.
+
```

Comment:
For personal preference, i think it would be better if you can put a link here to refer to the source of these pluses for more detailed information, unless you come up with all these yourself. Same thing applies to other sections of this book chapter. Although there is reference to the source of information at the end, I think it would be better if you can put the link in each section to direct the reader to more detailed info without them scrolling to the end and looking for the detailed information in an all-encompassing website e.g. the vue official website.

Reply:
Ok, will make changes accordingly. Thanks both of you for the advice!